src/csharp.ts

```diff
@@ -0,0 +1,90 @@
+export namespace CSharp {
+	export function parseClasses(documentText: string): ParsedClass[] {
+		let matches = documentText.matchAll(/public class (\w+)[^{]*{((?:[^}{]+|{(?:[^}{]+|{[^}{]*})*})*)}/g);
+		return Array.from(matches).filter(matchArray => matchArray[1]).map(matchArray => <ParsedClass>{
+			className: matchArray[1],
+			properties: parseProperties(matchArray[2])
+		});
+	}
+
+	function parseProperties(classContents: string): Property[] {
+		// TODO: This doesn't catch lists of nullable types
+		// TODO: Doesn't handle multiple parameters in generic, like Dictionary
+		let matches = classContents.matchAll(/public ([\w<>]+)(\??) (\w+)\s?{\s?get;\s?set;\s?}/g);
```

Comment:
To consider:
1. Spaces between the get and semi-colon
2. Custom getter/setter functions